aws-lambda-java-events/src/main/java/com/amazonaws/services/lambda/runtime/events/CodePipelineEvent.java

```diff
@@ -0,0 +1,156 @@
+ * and limitations under the License.
+ */
+
+package com.amazonaws.services.lambda.runtime.events;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+
+import java.io.Serializable;
+import java.util.List;
+
+/**
+ * Represents an CodePipeline event sent to Lambda.
+ * See: <a href="https://docs.aws.amazon.com/codepipeline/latest/userguide/actions-invoke-lambda-function.html">Invoke an AWS Lambda function in a pipeline in CodePipeline</a>
+ */
+@Data
+@Builder(setterPrefix = "with")
+@NoArgsConstructor
+@AllArgsConstructor
+public class CodePipelineEvent implements Serializable {
+    private static final long serialVersionUID = -4828716548429210697L;
+
+    @JsonProperty("CodePipeline.job")
+    private Job codePipelineJob;
+
+    @lombok.Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class Job implements Serializable {
+        private static final long serialVersionUID = 2211711169692638977L;
+
+        private String id;
+        private String accountId;
+        private Data data;
+    }
+
+    @lombok.Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class Data implements Serializable {
+        private static final long serialVersionUID = 8786599041834868262L;
+
+        private ActionConfiguration actionConfiguration;
+        private List<Artifact> inputArtifacts;
+        private List<Artifact> outputArtifacts;
+        private ArtifactCredentials artifactCredentials;
+        private String continuationToken;
+        private EncryptionKey encryptionKey;
+    }
+
+    @lombok.Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class ActionConfiguration implements Serializable {
+        private static final long serialVersionUID = -7285651174501621217L;
+
+        private Configuration configuration;
+    }
+
+    @lombok.Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class Configuration implements Serializable {
+        private static final long serialVersionUID = 580024317691702894L;
+
+        @JsonProperty("FunctionName")
+        private String functionName;
+        @JsonProperty("UserParameters")
+        private String userParameters;
+    }
+
+    @lombok.Data
+    @Builder(setterPrefix = "with")
+    @NoArgsConstructor
+    @AllArgsConstructor
+    public static class Artifact implements Serializable {
+        private static final long serialVersionUID = 6406621244704594358L;
+
+        private String name;
+        private String revision;
+        private Location location;
+
+        @JsonInclude
+        public String getRevision() {
```

Comment:
I added specific getter and setter for this attribute because Jackson was not including the attribute when it was attached to the parameter.

@andclt suspected it might be a problem with Lombok and Jackson not working together.